C_Flat_Interpreter/Parser/Parser.cs

```
@@ -27,799 +7,729 @@ namespace C_Flat_Interpreter.Parser;
+                    if (existingType != NodeType.Null && existingType != assignmentValue.type)
+                    {
+                        throw new SyntaxErrorException($"Invalid variable assignment! Type '{assignmentValue.type}' cannot be assigned to '{identifier}' of type '{existingType}'", _currentLine);
+                    }
+                }
+                else
+                {
+                    throw new OutOfScopeException($"Variable '{identifier}' does not exist in this context", _currentLine);
+                }
+            }
+            VariableTable.Add(identifier, assignmentValue);
+            //	Only add if the assignment is valid
+            node.AddChild(valueNode);
+        }
+        catch (InvalidSyntaxException)
+        {
+            throw new SyntaxErrorException($"Invalid assignment value, unable to parse variable assignment", _currentLine);
+        }
+
+        //	Finally check for termination
+        if (Match(TokenType.SemiColon))
+        {
+            node.AddChild(new ParseNode(NodeType.Terminal, _tokens[_currentIndex]));
+            Advance();
+        }
+        else
+        {
+            throw new SyntaxErrorException($"Unterminated assignment, expected ';'. Actual: '{_tokens.ElementAtOrDefault(_currentIndex)}'", _currentLine);
+        }
     }
 
-    #endregion
+    private void VariableAssignmentValue(ParseNode node)
+    {
+        //	Store current index
+        int index = _currentIndex;
+        
+        foreach (var assignmentValue in _variableAssignmentValueDictionary)
+        {
+            try
+            {
+                node.AddChild(CreateNode(assignmentValue.Key, assignmentValue.Value));
+                return;
+            }
+            catch (ParserException e)
+            {
+                if (e is SyntaxErrorException)
+                    throw;
+                Reset(index);
+                _logger.Debug(e.Message);
+            }
+        }
```

We need to throw an exception after the foreach, so that if we cannot parse an assignment it doesn't just return and instead throws.